tests/e2e/tests/controller/controller_test.go

```diff
@@ -59,7 +58,11 @@ func makeClient(t *testing.T, desc model.ConfigDescriptor) (*crd.Client, error)
 
 // makeTempClient allocates a namespace and cleans it up on test completion
 func makeTempClient(t *testing.T) (*crd.Client, string, func()) {
-	client, err := kube.CreateInterface(os.Getenv("KUBECONFIG"))
+	kubeconfig, err := kube.ResolveConfig("")
+	if err != nil {
+		t.Fatal(err)
+	}
+	client, err := kube.CreateInterface(kubeconfig)
```

Comment:
AFAICT, `kube.CreateInterface("")` should do the right thing here without `kube.ResolveConfig`. It uses k8s's helper function `clientcmd.BuildConfigFromFlags("", kubeconfig)` which should handle all the various incantations for sourcing kubeconfig (envvars, files, flags). We tried to fix all of these in https://github.com/istio/istio/pull/5566, but it looks like we missed the case here.

Reply:
If that was the case, then there would not be a bug. https://sourcegraph.com/github.com/istio/istio@master/-/blob/vendor/k8s.io/client-go/tools/clientcmd/client_config.go#L527 if not set it use inClusterConfig

Reply:
Since the change is only in the test, I'd not worry about it too much.

I'd like to get this in to unblock the release tests ASAP.

Reply:
+1, not worth blocking this PR on my question. I was more interested in why this isn't working since we use similar config resolution logic elsewhere for pilot, istioctl, etc. and it's important we get that right there.